render/gles2/texture.c

```diff
@@ -207,7 +207,8 @@ static void gles2_texture_get_matrix(struct wlr_texture_state *texture,
 	wlr_matrix_identity(matrix);
 	wlr_matrix_translate(&world, x, y, 0);
 	wlr_matrix_mul(matrix, &world, matrix);
-	wlr_matrix_scale(&world, _texture->width, _texture->height, 1);
+	wlr_matrix_scale(&world, _texture->width_from_buffer,
+		_texture->height_from_buffer, 1);
```

Comment:
Involving wlr_render in this seems like the wrong approach imo. Better to let the caller adjust the matrix as necessary. Maybe we shouldn't use this function at all and instead just let the caller sling matricies themselves? Or maybe we can just let the user pass in another matrix to multiply the world by.